src/documents/templates/DeploymentTemplateDoc.ts

```diff
@@ -477,10 +477,115 @@ export class DeploymentTemplateDoc extends DeploymentDocument {
                 actions.push(...scopeActions);
             }
         }
-
+        return this.addExtractCodeActions(range, actions);
+    }
+    private addExtractCodeActions(range: Range | Selection, actions: (CodeAction | Command)[]): (CodeAction | Command)[] {
+        let shouldAddExtractActions: boolean = false;
+        let pc = this.getContextFromDocumentLineAndColumnIndexes(range.start.line, range.start.character, undefined);
+        if (range.start.line === range.end.line) {
+            let jsonToken = pc.document.getJSONValueAtDocumentCharacterIndex(pc.jsonTokenStartIndex - 1, Span.ContainsBehavior.extended);
+            if (jsonToken instanceof Json.Property && pc.document.topLevelValue) {
+                let resources = pc.document.topLevelValue.getPropertyValue("resources");
```

Comment:
Since Extract Parameter/Variable are code actions I thought we would be restrictive when we should them or people might get annoyed (if they are shown when they do not make sense).

Some examples
* Inside a function (this is not supported)
* Extract variable from the default value of a parameter

We we remove the check we will probably run into a lot of edge cases that might annoy people. 
I am not sure it will be worth it comparing to people that want to extract a variable from a parameter or output

Reply:
Good for now.